prefect2/data_source_work_queues.go

```diff
@@ -0,0 +1,140 @@
+package prefect2
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	hc "terraform-provider-prefect2/prefect2_api"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceWorkQueues() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceWorkQueuesRead,
+		Schema: map[string]*schema.Schema{
+			"workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"work_queue_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"work_queues": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"created": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"updated": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"is_paused": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"concurrency_limit": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"priority": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"work_pool_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"filter": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"last_polled": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceWorkQueuesRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*hc.Client)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	workspaceId := d.Get("workspace_id").(string)
+	workQueueId := d.Get("work_queue_id").(string)
+
+	var workQueuesOutput []interface{}
+
+	if workQueueId != "" {
+		workQueue, err := c.GetWorkQueue(workQueueId, workspaceId)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		workQueues := make([]hc.WorkQueue, 1, 1)
```

Comment:
small nitpick (feel free to ignore, we can clean up later):

Manually specifying the length and capacity of array slices is not common in Go, except for cases requiring performance optimization by preallocating memory. Instead, the two preferred ways to do this in Go are to use `append`:

```go
	workQueues := []WorkQueue{}
	workQueues = append(workQueues, WorkQueue{
		Id:      "hello",
		Created: "world",
	})
```

or to define it as a literal, which I think is preferred in this case:

```go
	workQueues = []WorkQueue{
		{
			Id:      "hello",
			Created: "world",
		},
	}
```

You can see both examples here: https://go.dev/play/p/8eaiJ4PCk1J

I haven't tried it, but in this case, I think this should work:

```suggestion
		workQueues := []hc.WorkQueue{
			*workQueue,
		},
```

Reply:
Thanks for the suggestion. This has been changed.